ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/Keystore.hs

```diff
@@ -263,3 +210,7 @@ toList (Keystore ks) =
 
 walletIdToKey :: WalletId -> AddressHash PublicKey
 walletIdToKey (WalletIdHdSeq hdRootId) = view fromDb $ getHdRootId hdRootId
+
+-- | Whether stored InternalStorage has empty keys
+isEmptyInternalStorage :: InternalStorage -> Bool
```

Comment:
Tbh, I do not understand why `getInternalStorageKeys` is in `Types` and this function is here. Both are simple pure functions which take `InternalStorage` and return something, so I think they should be in the same module :man_shrugging:

Reply:
The reason is to avoid cyclic dependencies. An excellent place to put `getInternalStorageKeys` would be `Ariadne.Wallet.Cardano.Kernel.Keystore.Util` but then it would lead to cyclic dependencies since it would require `InternalStorage` in `Ariadne.Wallet.Cardano.Kernel.Keystore.Types` . `isEmptyInternalStorage` would also have a similar issue